src/test/py/bazel/bzlmod/bazel_lockfile_test.py

```diff
@@ -839,6 +839,74 @@ def testOldVersion(self):
       data = json.load(json_file)
     self.assertEqual(data['lockFileVersion'], 1)
 
+  def testExtensionEvaluationDoesNotRerunOnChangedImports(self):
+    self.ScratchFile(
+      'MODULE.bazel',
+      [
+        'lockfile_ext = use_extension("extension.bzl", "lockfile_ext")',
+        'use_repo(lockfile_ext, "dep", "indirect_dep", "invalid_dep")',
+      ],
+    )
+    self.ScratchFile('BUILD.bazel')
+    self.ScratchFile(
+      'extension.bzl',
+      [
+        'def _repo_rule_impl(ctx):',
+        '    ctx.file("WORKSPACE")',
+        '    ctx.file("BUILD", "filegroup(name=\'lala\')")',
+        '',
+        'repo_rule = repository_rule(implementation=_repo_rule_impl)',
+        '',
+        'def _module_ext_impl(ctx):',
+        '    print("I am being evaluated")',
+        '    repo_rule(name="dep")',
+        '    repo_rule(name="missing_dep")',
+        '    repo_rule(name="indirect_dep")',
+        '    return ctx.extension_metadata(',
+        '        root_module_direct_deps=["dep", "missing_dep"],',
+        '        root_module_direct_dev_deps=[],',
+        '    )',
+        '',
+        'lockfile_ext = module_extension(',
+        '    implementation=_module_ext_impl',
+        ')',
+      ],
+    )
+
+    # The build fails due to the "invalid_dep" import, which is not generated by the extension.
+    # Warnings should still be shown.
+    _, _, stderr = self.RunBazel(['build', '@dep//:all'], allow_failure = True)
+    self.assertIn('I am being evaluated', '\n'.join(stderr))
```

Comment:
nit: `stderr = '\n'.join(stderr)` here and below, just so that we don't do a big string concatenation 5 times every time